packages/semantic-model/src/convert.ts

```diff
@@ -193,7 +193,15 @@ function convertClass(
       symbol["ui5-metadata"].properties,
       partial(convertProperty, libName, clazz)
     );
+    // Add special settings (for example: "id" from ManagedObject)
+    clazz.properties = clazz.properties.concat(
```

Comment:
Should we add these properties on **every** class? or should we instead add these only on the top level class[es]?

@tal-sapan WDYT?

Reply:
They only exist on a handful of classes (like `ManagedObject`, which `Element` inherits from so all controls and elements inherit from transitively). So there's no need to check where it's coming from. They're like regular properties.

Reply:
But won't this mean we are breaking the model because, it may include classes that do not inherit from ManagedObject and yet we add these properties to **all** the classes...

Remember that we may want this model for other scenarios as well...

Reply:
We're not adding them on classes that don't have them... they will only be added for classes that have `specialSettings` under `ui5-metadata`

Reply:
Ah I see now.